### ISSUE TITLE
[Feature Request]: Auto Roll-Back

### ISSUE THREAD

**Issue by pixincreate** [topjohnwu/Magisk#3764]
Yep, you heard it right. Lemme explain in detail.   

# What's the problem now?  
- On every OTA update(A/B devices), slots are changed, boot images are backed up automatically. And that's not an issue.
- But, when ever an update to Magisk arrives, like in today's case(You merged both managerand Magisk), sometimes, the app hangs itself and doesn't responds which makes me(user) to re-start the app when the patching was going on.
- Which in-turn crashes boot.img(as well as backed up one.) allowing device not to boot up again. 
- If you don't have the boot.img available in **download section** as a _copy_, the only way to restore phone is by flashing(incremental OTA sucks tho!)  
    
## Ok, What I'm trying to ask?  
- An option that auto-rollbacks to the previous version when the device doesn't boots up, say, `bootloop` or `boot.img` lost.
- You can see this feature visible in some of the Magisk Modules like WebView Switcher, Riru modules, osm0sis and etc, you get my point.!
- So, basically, we need `stock boot.img` and `magisk patched boot.img` copies to be visible in the downloads section along with the magisk usual backup, and the above said roll-back option.
     
## Why we need copies?
Just for safety purposes, incase, something goes wrong, we'll have a backup on our hands.(You can provide that backup copy as a toggle button in the magisk manager so that, interested people can turn them on, others don't).
     
     
[Edit]: I'd to ask because, after the mentioned issue(under _Wha'ts the problem now?_), my phone could not boot up and I didn't even had/have the backup of `December Security Patched Stock Boot Image` and had to flash the entire device now(being flashed).  
     

_Thank you for having some patience to read._

**Comment by DanGLES3** [comment]
May I ask, how would that work? if the boot image is malformed to the point it doesn't boot how would Magisk (which depends on a functional boot image to runs its code) be able to  function in order to restore a backup?

**Comment by topjohnwu** [comment]
Sorry, but unfortunately this is technically impossible unless we modify the bootloader.

**Comment by pixincreate** [comment]
Oh, okay. But do you(@topjohnwu ) really think that implementing a method to backup `patched boot img` along with `stock boot img` and if anything goes wrong, either revert back to previous one or flash stock boot img. Then pop-up and notify the user that `something went wrong, root access removed, flash the previous working version until a fix is updated` and bla bla bla or etc.,    
Or, it'd be great if you put some light about where the backup of `stock boot img` exists after every OTA update. Because, you know, how good Nokia is.(Yes, I'm a Nokia phone user, sadly)
